controller/user-controller.js

```diff
@@ -0,0 +1,194 @@
+const jwt = require('jsonwebtoken');
+const bcrypt = require('bcrypt');
+const nodemailer = require('nodemailer');
+require('dotenv').config();
+const userService = require('../service/user-service');
+const tokenService = require('../service/token-service')
+
+
+// function to register user
+const register = async(req, res) => {
+
+    const {username, email, password, confirmPassword} = req.body;
+
+    if(username ==  "" || email === "" || password === "" || confirmPassword === ""){
```

Comment:
use proper condition for error handling(undefined case will not be handle by this cond)